react-validated-input-component/my-app/src/ValidatedInput.js

```diff
@@ -0,0 +1,40 @@
+import { useState } from "react";
+
+export default function ValidatedInput() {
+  const [pass, setPass] = useState('');
+  const [invalid, setInvalid] = useState(true)
+  const [text, setText] = useState('Password is required.')
+  const [icon, setIcon] = useState('')
```

Comment:
Many of these can be computed directly from `pass`. When you can do that, it's best to not use state.